bin/common.sh

```diff
@@ -106,7 +106,13 @@ function getZeppelinVersion(){
     exit 0
 }
 
-# Text encoding for 
+function downloadSparkBinary() {
+  if [[ -z "${SPARK_HOME}" ]]; then
```

Comment:
One possible issue is
SPARK_HOME can not only be defined in `zeppelin-env.sh` but also in GUI interpreter setting page (related bug is being handled by [ZEPPELIN-1334](https://issues.apache.org/jira/browse/ZEPPELIN-1334)).

If SPARK_HOME is not defined in `zeppelin-env.sh` but in interpreter setting page on GUI, this script will not recognize SPARK_HOME at this point. Because `downloadSparkBinary()` is invoked by either `bin/zeppelin-daemon.sh` or `bin/zeppelin.sh`, but SPARK_HOME defined from GUI will be propagated to `bin/interpreter.sh` only.

Another possible issue is
Some user may doesn't even have a spark interpreter installed. (e.g. netinst package and user installed jdbc interpreter only) In this case downloading spark binary because of SPARK_HOME is not defined doesn't make any sense.

Considering these two possible issue around this conditional statement, i suggest change this condition to check installation of spark interpreter, not check SPARK_HOME. for example

```
if [[ -d "${ZEPPELIN_HOME}/interpreter/spark" ]]; then
```

And inside of `bin/download-spark.sh` script, how about just ask user explicitly whether user want to download spark binary for local mode or not? And if user answers 'N' we can create a small text file somewhere under `interpreter/spark` to remember the answer.

What do you think?